blueprints/materials/concrete.py

```diff
@@ -1 +1,547 @@
+
+        Returns
+        -------
+        CementClass
+            Example: "N" (standard value)
+        """
+        return self._cement_class
+
+    @property
+    def density(self) -> KG:
+        """Unit mass of concrete [kg/m3].
+
+        Returns
+        -------
+        KG
+            Example: 2500.0 (for regular reinforced concrete)
+        """
+        return self._density
+
+    @property
+    def diagram_type(self) -> DiagramType:
+        """Type of stress-strain diagram (Figures 3.2, 3.3 and 3.4 from NEN-EN 1992-1-1).
+
+        Returns
+        -------
+        DiagramType
+            Example: DiagramType.BI-LINEAR
+        """
+        return self._diagram_type
+
+    @property
+    def aggregate_type(self) -> ConcreteAggregateType:
+        """Type of aggregate in the concrete material (NEN-EN 1992-1-1).
+
+        Returns
+        -------
+        ConcreteAggregateType
+            Example: "Kwartsiet" (standard value)
+        """
+        return self._aggregate_type
+
+    @property
+    def aggregate_size(self) -> MM:
+        """Largest nominal maximum aggregate size [mm] (NEN-EN 1992-1-1).
+
+        Returns
+        -------
+        MM
+            Example: 16.0 (standard value)
+        """
+        return self._aggregate_size
+
+    @property
+    def plain_concrete_diagram(self) -> bool:
+        """Use a Stress-strain diagram for plain or lightly reinforced concrete (NEN-EN 1992-1-1). (True or False).
+
+        Returns
+        -------
+        bool
+            Example: False (standard value)
+        """
+        return self._plain_concrete_diagram
+
+    def rho_min(self, f_yd: MPA) -> PERCENTAGE:
+        """Minimum reinforcement ratio (CB2, 7de druk 2011, pag.55) [%].
+
+        Parameters
+        ----------
+        f_yd: MPA
+            Design yield strength of reinforcement [MPa]
+
+        Returns
+        -------
+        PERCENTAGE
+        """
+        return (0.223 * (self.f_ctm / f_yd)) * 100
+
+    @property
+    def cement_type(self) -> CementType:
+        """Type of cement in the concrete material (NEN-EN 1992-1-1).
+
+        Returns
+        -------
+        CementType
+            Example: CementType.CEM_I
+        """
+        return self._cement_type
+
+    def __eq__(self, other: object) -> bool:
+        """Check if two ConcreteMaterial objects are equal."""
+        if not isinstance(other, ConcreteMaterial):
+            raise NotImplementedError
+        properties_self = {
+            attribute: self.__getattribute__(attribute)
+            for attribute in dir(self)
+            if not attribute.startswith("_") and attribute not in ["id", "name", "rho_min"]
+        }
+        properties_other = {
+            attribute: other.__getattribute__(attribute)
+            for attribute in dir(self)
+            if not attribute.startswith("_") and attribute not in ["id", "name", "rho_min"]
+        }
+        return properties_self == properties_other
```

Comment:
I don't get why we don't use the dataclass implementation, as I see now: this class has the dataclass decorator, but doesn't really use it's power. 

Alternatively If we would like to do it this way, there is a much cleaner implementation --> implement a '__key' method which returns a tuple with the attributes you want to compare and use that in this eq. method (and lose these DRY) --> IMO the eq. method shouldn't be responsible for creating such data.

Reply:
a custom __eq__ is not necessary anymore. We are using a proper dataclass now.